### PR TITLE
CSR representation added(still working on bfs on CSR)

### PR DESCRIPTION
**Pull request by barangayretli**

### CODE REVIEW

File: graph.cpp [hunk 1]
```diff
@@ -231,3 +235,103 @@ void adjListVect::printArr(vector<string> arr)
...
+        currIndex+=v.adjVect[i].second.size();
+    }
+    
+}
+
+void CSR::buildCSR(adjListVect v)
+{
+    for(int i=0; i<v.adjVect.size();i++)
+    {
+        CSRmatrix.push_back(mapValues[v.adjVect[i].first]);
```
Comment (barangayretli) [inline graph.cpp]: this line will be deleted

**Comment by anilpacaci** [comment]
Recommended Changes
- Delete the RPQ folder
- Create a separate header for each main class (automate, graph ...)

**Comment by anilpacaci** [comment]
Templates:
You can have one template class Graph with methods:
- addEdge
- getNeighours
...

then both graph and ProductGraph are instances of this parametrized class with different template parameters.

File: graph.h [hunk 2]
```diff
@@ -61,9 +61,28 @@ class productGraph{
...
     void buildPG(graph g, automata q);
     void results(string edge1);
     void printArr(vector<string> arr);
 };
+
+class CSR{
+private:
+    int currIndex=0;
+    int count=0;
+    vector<int> CSRmatrix;
```
Comment (anilpacaci) [inline graph.h:76]: since we know how many vertices and edges are there, you can simply use int arrays to represent these two

**Pull request merged**